docker/orb.yaml

```diff
@@ -362,6 +362,33 @@ commands:
               docker push "<<parameters.registry>>/<<parameters.image>>:${CIRCLE_TAG}"
             fi
 
+  push-with-custom-tag:
+    description: >
+      Push the built docker image with the provided custom tag.
+    parameters:
+      image:
+        default: ${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}
+        description: >
+          Name of the target image. Note that this includes any namespacing --
+          eg. for pushing to Dockerhub, you likely want to ensure this looks
+          like "talkiq/foobar"; for other repositories, this may be some
+          other namespace such as "my-gcp-project/my-image-name" for GCR.
+        type: string
+      registry:
+        default: 'docker.io'
+        description: >
+          Container registry to-be-used.
+        type: string
+      tag:
```

Comment:
Note that it could be valuable to make this a feature of `push-with-tag` rather than an entirely new command. You could have `tag` get a default value of `$CIRCLE_TAG`, which would let `push-with-tag` keep the same default behaviour but also allow users to override it with a specified string as need be.

Reply:
Good call that's definitely the better solution :)